src/control/server/ctl_system.go

```diff
@@ -231,6 +233,31 @@ func (svc *ControlService) prepShutdown(ctx context.Context, rankList []system.R
 	return results, nil
 }
 
+// resultsFromUnreachableHosts generates member results for ranks on unreachable hosts.
+func (svc *ControlService) resultsFromUnreachableHosts(hostRanks map[string][]system.Rank, hostErrors control.HostErrorsMap) system.MemberResults {
+	ranks := make([]system.Rank, 0, len(hostRanks)*maxIOServers)
+
+	// add rank results for any on reachable harness addrs
+	for errMsg, hostSet := range hostErrors {
+		if !strings.Contains(errMsg, "connection refused") {
+			continue
+		}
+		for _, addr := range strings.Split(hostSet.DerangedString(), ",") {
+			ranks = append(ranks, hostRanks[addr]...)
+			svc.log.Debugf("no response from harness %s", addr)
+		}
+	}
```

Comment:
I don't understand why it matters what the error was. Shouldn't we just be excluding all ranks that are associated with a host error from being in the stopped state?

Reply:
the thinking is that there could be another source of error that doesn't indicate that the rank's process is down necessarily, at this point I'm happy to assume a rank is down if any host error is returned.